ldms/src/ldmsd/ldmsd.c

```diff
@@ -103,6 +103,7 @@
 #define LDMSD_MEM_SIZE_ENV "LDMSD_MEM_SZ"
 #define LDMSD_MEM_SIZE_STR "512kB"
 #define LDMSD_MEM_SIZE_DEFAULT 512L * 1024L
+#define LDMSD_FLUSH_LOG_TIME 10 * 60 /* 10 mins */
```

Comment:
@nichamon, I don't see why we need this capability. If the log message rates are low, the log buffer will get flushed because the log event queue is often empty. If the log message rates are high, the log buffer (4096B) will fill up and get flushed anyway by libc. Am I missing something?

Reply:
@tom95858 No, you didn't miss anything. I added the code to make a comparison without the patch. However,  OVIS-4 doesn't use the event-driven paradigm, so the only way is to flush periodically. I just tried running the current OVIS-4 with the INFO log level on Voltrino. With the OVIS-4 code case, Zap spawned tens of threads because the threads were stuck in flushing the log file.

Reply:
Moreover, I have a concern that the condition `0 == ev_pending(logger_worker)` may flush too often. I would like to compare the memory usages of the condition and the fixed interval strategy. The strategy that flushes when the queue is empty may have worse performance than the fixed interval in the situation that most log messages enqueue right after the logger worker processes the previous log message.